magicctapipe/scripts/lst1_magic/lst1_magic_stereo_reco.py

```diff
@@ -377,6 +382,30 @@ def stereo_reconstruction(input_file, output_dir, config, magic_only_analysis=Fa
         event_data, output_file, group_name="/events", table_name="parameters"
     )
 
+    if save_images:
+        tel_ids = np.unique(event_data["tel_id"]).tolist()
+        for tel_id in tel_ids:
+            key = f"/events/dl1/image_{tel_id}"
+            image = read_table(input_file, key)
+            mask = np.zeros(len(image), dtype=bool)
+            tag = "lst" if tel_id in LSTs_IDs else "magic"
+            obs_evt_ids = event_data.query(f"tel_id=={tel_id}")[
+                [f"obs_id_{tag}", f"event_id_{tag}"]
+            ].to_numpy()
+            for obs_id, evt_id in obs_evt_ids:
+                this_mask = np.logical_and(
+                    image["obs_id"] == obs_id, image["event_id"] == evt_id
+                )
+                mask = np.logical_or(mask, this_mask)
```

Comment:
It feels like there should be a better way than to loop other all obs_id, evt_id. But I don't have a solution yet.

Reply:
I know what you mean, but I was also not able to find a better way for it